src/main/scala/scalismo/registration/PointwiseHuberLossMetric.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2015 University of Basel, Graphics and Vision Research Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package scalismo.registration
+
+import scalismo.geometry.{ Dim, NDSpace }
+import scalismo.image.{ DifferentiableScalarImage, ScalarImage }
+import scalismo.numerics._
+import scalismo.utils.Random
+
+/**
+ * Image to image metric which applies the Huber Loss function to the pointwise pixel difference.
+ * The parameter delta defines the threshold. The Huber loss increases quadratically for
+ * values below this threshold and linearly for values above this threshold.
+ * @see SumOfPointwiseLossMetric.
+ *
+ */
+case class PointwiseHuberLossMetric[D <: Dim: NDSpace](fixedImage: ScalarImage[D],
+  movingImage: DifferentiableScalarImage[D],
+  transformationSpace: TransformationSpace[D],
+  sampler: Sampler[D],
+  delta: Double = 1.345)(implicit rng: Random)
+    extends SumOfPointwiseLossMetric(fixedImage, movingImage, transformationSpace, sampler) {
```

Comment:
Maybe the MeansquaresMetric should also be brought under this hierarchy then?

Reply:
If we would define as the loss function the squared function, we would get the sum of squared differences metric. This is slightly different from the mean squares metric, which takes the average over all the points.

But you are raising a very good point. Maybe we should also average the loss in the new metric, as otherwise the value of the loss function depends on the number of points that are sampled? This would mean that the more points we sample, the larger the weight of the metric compared to the regularizer.

Reply:
That would make a lot of sense, yes. In which case, the Meansquares would also be under that hierarchy.

Reply:
I just realized that I was wrong. The averaging is already done by the integrator. Strangely enough, the integrator takes the number of points, but not the volume into account. This might actually be a bug in the intergrator ...

I will put the mean squares metric under this scheme here and also check what should be the correct behavior of the integrator.